src/lib/customPropTypes.js

```diff
@@ -181,6 +181,49 @@ export const demand = (requiredProps) => {
 }
 
 /**
+ * Ensure a component can render as a node passed as a prop value in place of children.
+ */
+export const content = (...args) => every([
+  disallow(['children']),
+  PropTypes.node,
+])(...args)
+
+/**
+ * Item shorthand is a description of a component that can be a literal,
+ * a props object, or an element.
+ */
+export const item = (...args) => every([
+  disallow(['children']),
+  PropTypes.oneOfType([
+    PropTypes.element,
+    PropTypes.number,
+    PropTypes.object,
+    PropTypes.string,
+  ]),
+])(...args)
+
+/**
+ * ArrayOf shorthand ensures a prop is an array of item shorthand
+ */
+export const items = (...args) => every([
+  disallow(['children']),
```

Comment:
I'm trying to unwind this a little bit because it's gets a little complicated with the nested `every` calls. Turns out this `disallow` is actually necessary, I'll try to explain. Consider the following:
```js
const cells = ['Name', 'Status', 'Notes']
<TableRow cells={cells}>some child</TableRow>
```

If cells were `PropTypes.arrayOf(item)` this doesn't give a warning because `arrayOf` will use the given validation (`item` in this case) to validate each item in the `cells` within the context of `cells`, not within the context of `TableRow`.

Nested `disallow` doesn't actually seem to work as expected since it operates on the parent context of the given prop. So when checking `cells` with:
```js
const items = (...args) => every([
  disallow(['children']),
  PropTypes.arrayOf(item),
])(...args)
```
It's evaluating `cells` within the context of the TableRow props and asking **"is cells an array of items AND is the children key undefined _in TableRow props_"**. When it's evaluating each item via `arrayOf(item)`, it's asking **"is this item one of these valid types AND is the children key undefined _in this item's array_"**  (i.e. `array['children']`) which doesn't throw but doesn't really make sense 

I hope that makes sense but it's pretty complicated haha.